vector/src/main/java/im/vector/app/features/userdirectory/UserListViewModel.kt

```diff
@@ -241,11 +241,11 @@ class UserListViewModel @AssistedInject constructor(
                         .searchUsersDirectory(search, 50, state.excludedUserIds.orEmpty())
                         .sortedBy { it.toMatrixItem().firstLetterOfDisplayName() }
                 val userProfile = if (MatrixPatterns.isUserId(search)) {
-                    val json = tryOrNull { session.getProfile(search) }
+                    val user = tryOrNull { session.getProfileAsUser(search) }
                     User(
                             userId = search,
-                            displayName = json?.get(ProfileService.DISPLAY_NAME_KEY) as? String,
-                            avatarUrl = json?.get(ProfileService.AVATAR_URL_KEY) as? String
+                            displayName = user?.displayName,
+                            avatarUrl = user?.avatarUrl
```

Comment:
You don't have to create a new User object if you already have one, just handle the nullability case:
`tryOrNull { session.getProfileAsUser(search) } ?: User(search)`

Reply:
nice catch !